File: controllers/UnsplashController.cs
using System.Linq;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using OpenAI_API;

[ApiController]
[Route("api/[controller]")]
public class UnsplashController : ControllerBase
{
    private readonly UnsplashService _unsplashService;

    public UnsplashController(UnsplashService unsplashService)
    {
        _unsplashService = unsplashService;
    }

    [HttpPost("search")]
    public async Task<IActionResult> SearchSneakerImages([FromBody] SearchRequest request)
    {
        var images = await _unsplashService.SearchSneakerImages(request.Descricao, request.Cor, request.Estilo);
        return Ok(new { images });
    }
}

public class SearchRequest
{
    public string Descricao { get; set; }
    public string Cor { get; set; }
    public string Estilo { get; set; }
}

File: controllers/unsplashController.js
const axios = require('axios');
const { Configuration, OpenAIApi } = require('openai');

const configuration = new Configuration({
  apiKey: process.env.OPENAI_API_KEY,
});
const openai = new OpenAIApi(configuration);

const resumirDescricao = async (descricao) => {
  try {
    const response = await openai.createCompletion({
      model: "text-davinci-003",
      prompt: `Resuma isso em 4 palavras: "${descricao}"`,
      max_tokens: 10,
    });
    return response.data.choices[0].text.trim();
  } catch (error) {
    console.error('Erro ao resumir descrição:', error);
    return descricao; // Retorna a descrição original em caso de erro
  }
};

const searchImages = async (req, res) => {
  const { descricao, cor, estilo } = req.body;
  try {
    const response = await axios.get('https://api.unsplash.com/search/photos', {
      params: { query: descricao, color: cor, orientation: estilo },
      headers: {
        Authorization: `Client-ID ${process.env.UNSPLASH_ACCESS_KEY}`,
      },
    });

    const images = await Promise.all(response.data.results.map(async (img) => {
      const resumoDescricao = await resumirDescricao(img.description || img.alt_description || '');
      return {
        ...img,
        resumoDescricao,
      };
    }));

    res.json({ images });
  } catch (error) {
    console.error('Erro ao buscar imagens:', error);
    res.status(500).json({ error: 'Erro ao buscar imagens' });
  }
};

module.exports = { searchImages };